channeldb/channel.go

```diff
@@ -2203,6 +2224,32 @@ func deserializeCloseChannelSummary(r io.Reader) (*ChannelCloseSummary, error) {
 		}
 	}
 
+	// Check if we have a channel sync message to read.
+	var hasChanSyncMsg bool
+	err = ReadElements(r, &hasChanSyncMsg)
+	if err == io.EOF {
```

Comment:
Note: we can add write the `false` marker here as part of the migration to avoid this `EOF` check.

Reply:
Currently the addition of the `LastChanSyncMsg` field is kept in its own commit to show how new fields can easily be added after the presence flags are introduced in the serialization.

Reply:
Why are we mixing the two approaches? We should either use EOF everywhere (so no migration) or the bool based approach.

Reply:
Ah scratch that, I see the usage now

Reply:
This is how you would add new fields in the future without a migration.